<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GermCellSimulator</class>
 <widget class="QMainWindow" name="GermCellSimulator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>862</width>
    <height>783</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>Germ Cell Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayoutMain">
      <item row="1" column="0">
       <layout class="QGridLayout" name="gridLayoutVisualSide">
        <item row="0" column="1">
         <widget class="QSlider" name="verticalSliderZoom">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <layout class="QGridLayout" name="gridLayoutVisualPlusMinus">
          <item row="1" column="0">
           <widget class="QLabel" name="labelPlus">
            <property name="font">
             <font>
              <family>Lucida Grande</family>
              <pointsize>18</pointsize>
              <weight>75</weight>
              <italic>false</italic>
              <bold>true</bold>
             </font>
            </property>
            <property name="contextMenuPolicy">
             <enum>Qt::NoContextMenu</enum>
            </property>
            <property name="text">
             <string>+</string>
            </property>
            <property name="textFormat">
             <enum>Qt::PlainText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignHCenter|Qt::AlignTop</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelMinus">
            <property name="font">
             <font>
              <family>Lucida Grande</family>
              <pointsize>18</pointsize>
              <weight>75</weight>
              <italic>false</italic>
              <bold>true</bold>
             </font>
            </property>
            <property name="contextMenuPolicy">
             <enum>Qt::NoContextMenu</enum>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="textFormat">
             <enum>Qt::PlainText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignBottom|Qt::AlignHCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <widget class="QGraphicsView" name="graphicsViewVisual">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="cursor" stdset="0">
           <cursorShape>ArrowCursor</cursorShape>
          </property>
          <property name="focusPolicy">
           <enum>Qt::StrongFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="interactive">
           <bool>true</bool>
          </property>
          <property name="dragMode">
           <enum>QGraphicsView::ScrollHandDrag</enum>
          </property>
          <property name="resizeAnchor">
           <enum>QGraphicsView::AnchorUnderMouse</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayoutVisualTop">
        <item row="1" column="4">
         <widget class="QLCDNumber" name="lcdNumberHour">
          <property name="font">
           <font>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="labelHour">
          <property name="font">
           <font>
            <pointsize>15</pointsize>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="text">
           <string>Hour</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <property name="margin">
           <number>3</number>
          </property>
          <property name="indent">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLCDNumber" name="lcdNumberDay">
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>1</number>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="labelDay">
          <property name="font">
           <font>
            <pointsize>15</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="text">
           <string>Day</string>
          </property>
          <property name="margin">
           <number>3</number>
          </property>
          <property name="indent">
           <number>-1</number>
          </property>
         </widget>
        </item>
        <item row="1" column="5">
         <spacer name="horizontalSpacerVisualTopRight">
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <spacer name="horizontalSpacerVisualTopLeft">
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="6">
         <widget class="QLabel" name="labelZoom">
          <property name="font">
           <font>
            <pointsize>13</pointsize>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="text">
           <string>Zoom</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="margin">
           <number>6</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QGridLayout" name="gridLayoutBottom">
      <item row="0" column="3">
       <widget class="QSlider" name="horizontalSliderSpeed">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="sliderPosition">
         <number>1000</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedAppearance">
         <bool>true</bool>
        </property>
        <property name="invertedControls">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="labelSpeed">
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="text">
         <string>Simulation Speed</string>
        </property>
        <property name="indent">
         <number>30</number>
        </property>
        <property name="buddy">
         <cstring>horizontalSliderSpeed</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="pushButtonPause">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="text">
         <string>Pause</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="pushButtonStart">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="pushButtonRecordStart">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Start Recording</string>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <widget class="QPushButton" name="pushButtonRecordStop">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Stop Recording</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>862</width>
     <height>23</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="contextMenuPolicy">
     <enum>Qt::NoContextMenu</enum>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_New"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionSave_CSV"/>
    <addaction name="actionSave_Simulator"/>
    <addaction name="actionSave_Position_Data"/>
    <addaction name="actionSave_Errors"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_New">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>Create a new simulation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionSave_CSV">
   <property name="text">
    <string>Save Population Stats</string>
   </property>
  </action>
  <action name="actionSave_Simulator">
   <property name="text">
    <string>Save Population Snapshot</string>
   </property>
  </action>
  <action name="actionSave_Position_Data">
   <property name="text">
    <string>Save Position Data</string>
   </property>
  </action>
  <action name="actionSave_Errors">
   <property name="text">
    <string>Save Errors</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonStart</sender>
   <signal>clicked()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>startSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>769</y>
    </hint>
    <hint type="destinationlabel">
     <x>8</x>
     <y>707</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonPause</sender>
   <signal>clicked()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>stopSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>164</x>
     <y>769</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>714</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderSpeed</sender>
   <signal>valueChanged(int)</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>changeSpeed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>582</x>
     <y>762</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>746</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalSliderZoom</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>changeZoom(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>794</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>803</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_New</sender>
   <signal>triggered()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>loadSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>375</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_CSV</sender>
   <signal>triggered()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>saveCSV()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Simulator</sender>
   <signal>triggered()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>saveData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Position_Data</sender>
   <signal>triggered()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>savePosData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonRecordStart</sender>
   <signal>clicked()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>startRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>640</x>
     <y>732</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonRecordStop</sender>
   <signal>clicked()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>stopRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>780</x>
     <y>732</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Errors</sender>
   <signal>triggered()</signal>
   <receiver>GermCellSimulator</receiver>
   <slot>saveErrors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>loadSimulation()</slot>
  <slot>startSimulation()</slot>
  <slot>stopSimulation()</slot>
  <slot>changeZoom(int)</slot>
  <slot>changeSpeed(int)</slot>
  <slot>saveCSV()</slot>
  <slot>saveData()</slot>
  <slot>loadData()</slot>
  <slot>savePosData()</slot>
  <slot>startRecording()</slot>
  <slot>stopRecording()</slot>
  <slot>saveErrors()</slot>
 </slots>
</ui>
